actors.go

```diff
@@ -37,3 +37,38 @@ type SignalError struct {
 func (e SignalError) Error() string {
 	return fmt.Sprintf("received signal %s", e.Signal)
 }
+
+// Upgrader handles zero downtime upgrades and passing files between processes.
+//
+// Upgrader is based on https://github.com/cloudflare/tableflip.
+type Upgrader interface {
+	Ready() error
+	Exit() <-chan struct{}
+	Stop()
+}
+
+// GracefulRestart returns an actor, i.e. an execute and interrupt func, that
+// terminates when graceful restart is initiated and the child process
+// signals to be ready, or the parent context is canceled.
```

Comment:
I'm not sure the description really matches the behavior. The Upgrader function really has nothing to do with upgrading; it's behavior is IMO better spelled as

```go
type StartStopWaiter interface {
    Start() error // start something running asynchronously
    Stop()       // stop what was previously started
    Wait()       // block until the started thing has been stopped
}
```

I understand the interface models a concrete type from another package, but upon reflection, I'm not sure how much sense it makes to bring specific integrations like this into the core package run. It feels like this is a totally reasonable amount of "glue" code that could/should live in whatever root package is importing both cloudflare/tableflip and oklog/run...

Reply:
Sounds good to me. The only thing I would keep is Wait returning a channel, so that we can select on `ctx.Done()` as well. If we want to keep the context around.

Reply:
I think I'm saying that it doesn't make sense to bring this into package run directly, because it's IMO a totally appropriate amount of glue code for consumers to write, and insufficiently general to be widely useful.

Reply:
(Not that I don't appreciate the thought and effort.)

Reply:
I see. Thanks for considering it.